File: 3634-MinimumRemovalstoBalanceArray/3634-MinimumRemovalstoBalanceArray.py
# Last updated: 2/7/2026, 12:20:24 AM
1class Solution:
2    def minRemoval(self, nums: List[int], k: int) -> int:
3        nums.sort()
4        n = len(nums)
5        i = 0
6        maxWindow = 0
7
8        for j in range(n):
9            while(nums[j] > k * nums[i]):
10                i += 1
11            maxWindow = max(maxWindow, j - i + 1)
12
13        return n - maxWindow

File: 3634-MinimumRemovalstoBalanceArray/3634-MinimumRemovalstoBalanceArray.ts
// Last updated: 2/6/2026, 11:57:29 PM
1function minRemoval(nums: number[], k: number): number {
2    /**
3        An array is considered balanced if the value of its maximum element is 
4        at most k times the minimum element.
5
6        I think this is straightforward.
7     */
8
9
10    nums.sort((a, b) => a - b);
11    let n = nums.length;
12    let i = 0;
13    let maxWindow = 0;
14
15
16    for (let j = 0; j < n; j++) {
17        while (nums[j] > k * nums[i]) {
18            i++;
19        }
20
21        maxWindow = Math.max(maxWindow, j - i + 1);
22    }
23    
24    return n - maxWindow;
25};